fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1301,15 +1301,19 @@ def _remove_shard_bwd_hook(fsdp_module: FullyShardedDataParallel) -> None:
             if isinstance(m, FullyShardedDataParallel):
                 _remove_shard_bwd_hook(m)
                 m._pre_backward_hook_has_run = False
-                if m._has_params:
-                    if any(p.requires_grad for p in m.params):
+                # Note: m.parameters() should not be an empty list. FSDP
+                # wrapping modules without weights is not tested at the moment.
```

Comment:
Should we add an error check here for this case then?

Reply:
Thanks for the review! I prefer to do that separately since it is a different use case and the check might be better in the init. I will file an issue.

Reply:
issue filed.